docs/reference/field-types/checkboxes.md

```diff
@@ -63,16 +71,68 @@ The `choices` setting in `checkboxes`, `radio`, or `select` fields configures th
 
 ## Populating `choices` dynamically
 
-What if the choices aren't known in advance? Then you can fetch them dynamically.
+What if the choices aren't known in advance or are dependent on the value of another schema field? Then you can fetch them dynamically.
 
-First, set the `choices` option to the name of a [method in your module](../module-api/module-overview.md#methods-self). Pass a string, the name of the method you'll implement on the server side — **do not** pass a function.
+First, set the `choices` option to the name of a [method in your module](../module-api/module-overview.md#methods-self). Pass the name of the method you'll implement on the server side — **do not** pass a function, as a string ending in `()`. e.g. `choices: 'getChoices()'`
 
-Second, implement that method in your module so that it takes `(req, data)` arguments and return an array of choices in the usual format. You may use an async function, or return a promise that will resolve to the array. That means you can reach out to APIs using modules like `axios` or `node-fetch`, or make Apostrophe database queries.
+Second, implement that method in your module so that it takes `(req, data, following)` arguments and return an array of choices in the usual format. You may use an async function, or return a promise that will resolve to the array. That means you can reach out to APIs using modules like `axios` or `node-fetch`, or make Apostrophe database queries.
 
-The `data` argument is an object containing the parent's `docId` for further inspection by your function.
+The `data` argument is an object containing the parent's `docId` for further inspection by your function and is falsey if the document hasn't been published.
+
+The `following` argument is an object containing the schema fields being followed as keys and the value of those fields as values. Note that the key will match the field name(s) in the `following` field array exactly. So if you are passing a parental schema field value in an `object` or `array` schema field, you need to include the prefixing `<` along with the name. See the [`following`](#following) property description.
 
 It is usually a good idea to perform at least short-term caching in your choices method, in order to limit the impact on performance when editing.
 
+``` javascript
+module.exports = {
+  extend: '@apostrophecms/piece-type',
+  options: {
+    label: 'Example'
+  },
+  fields: {
+    add: {
+      // other schema fields
+      region: {
+        type: 'select',
+        label: 'Region',
+        choices: [
+          { label: 'North America', value: 'na' },
+          { label: 'Europe', value: 'eu' }
+        ]
+      },
+      cities: {
+        type: 'checkboxes',
+        label: 'Favorite Cities',
+        following: [ 'region' ],
+        choices: 'getCities()'
+      }
+    }
+  },
+  methods(self) {
+    return {
+      getCities(req, { docId }, { region }) {
```

Comment:
I would add a comment here reiterating that this may be an async function

Reply:
Changed in all three files.